src/Api/DomainRecord.php

```diff
@@ -108,38 +108,30 @@ public function create($domainName, $type, $name, $data, $priority = null, $port
     }
 
     /**
-     * @param string $domainName
-     * @param int    $recordId
-     * @param string $name
-     * @param string $data
-     * @param int    $priority
-     * @param int    $port
-     * @param int    $weight
+     * @param string      $domainName
+     * @param int         $recordId
+     * @param string|null $name
+     * @param string|null $data
+     * @param int|null    $priority
+     * @param int|null    $port
+     * @param int|null    $weight
      *
      * @throws HttpException
      *
      * @return DomainRecordEntity
      */
     public function update($domainName, $recordId, $name = null, $data = null, $priority = null, $port = null, $weight = null)
     {
-        $content = array();
-        if ($name !== null) {
-            $content['name'] = $name;
-        }
-        if ($data !== null) {
-            $content['data'] = $data;
-        }
-        if ($priority !== null) {
-            $content['priority'] = $priority;
-        }
-        if ($port !== null) {
-            $content['port'] = $port;
-        }
-        if ($weight !== null) {
-            $content['weight'] = $weight;
-        }
-
-        return $this->updateFields($domainName, $recordId, ['name' => $name]);
+        $content = array(
+            'name' => $name,
+            'data' => $data,
+            'priority' => $priority,
+            'port' => $port,
+            'weight' => $weight
+        );
+        $content = array_filter($content);
```

Comment:
Please pass a callback function into here so that null values are filtered only, instead of just empty ones.